client/src/modules/reports/generate/debtorBalanceReport/debtorBalanceReport.js

```diff
@@ -0,0 +1,56 @@
+angular.module('bhima.controllers')
+.controller('debtorBalanceReportController', DebtorBalanceReportController);
+
+DebtorBalanceReportController.$inject = [
+  '$sce', 'NotifyService', 'BaseReportService', 'AppCache', 'BaseReportService', 'reportData',
+];
+
+function DebtorBalanceReportController($sce, Notify, BaseReportService, AppCache, SavedReports, reportData) {
+  var vm = this;
+  var cache = new AppCache('configure_debtorAccountBalence');
+  var reportUrl = 'debtorAccountBalence';
+
+  vm.reportDetails = {};
+  vm.currentFiscalYear = {
+    label : 'Test Fiscal Year 2015',
+  };
+
+  var _url = '';
+
+  vm.onSelectFiscal = function (fiscal) {
+    vm.currentFiscalYear.id = fiscal;
+    _url = reportUrl + '/' + vm.currentFiscalYear.id + '/'+ vm.currentFiscalYear.label;
```

Comment:
Why do you need to send back the label?

Reply:
It should appear in the report, as I can already have it from  a fiscal year component. I don't think it's important to execute another query to get the fiscal year label from it id